### ISSUE TITLE
Have plundering ships take time

### ISSUE THREAD

**Issue by wargwargwarg** [endless-sky/endless-sky#871]
Looting disabled cruisers is very unbalanced (in about 1 hour one can gather about 15Milion looting the small to medium cruisers, in a "good system with many pirates and many lawful ships to do the hard work). If you have a cavernous Mule it's much too easy to earn a ton of money at no risk.

This has been touched by several posts. Some say include wear for forcefully removing the components, others drastically drop the prices of used components.

 I propose a procedural approach, perhaps easier to do (it only needs the file with ship components)

Say you disabled a ship with 10 systems installed and 5 crew onboard. It is drifting, with 40% hull (either powerplant or engines or both are damaged). The 5 crew defend the critical systems, like powerplant and thrusters, in the hope of repairing them and getting out of there.

40% hull would mean 6 out of 10 systems are damaged in some way, on a random basis and may or may not be worth removing. At least one system should be the powerplant or thrusters.
The 5 crew defend the 2 systems ( incidentally, the most lucrative) .

In boarding panel, the list details what is damaged (perhaps degree, estimated value), what is intact and ready for looting (again value) and what is defended (by how many each, value, damage if any). And you choose if you loot, drive away defenders and loot that system or attempt capture.

More, all this looting is being done on the run, with hostiles onboard and hostile ships in the area. There should be a time counter ("you have 1 hour to grab what you can"), the limitating factor being that the more mass the device has, the more time it takes to bring it to your hold). There is no time to "loot all their shit" (Firefly, anyone?).

PS. Much simpler (and very unpopular, I guess) downgrade the Mule

**Comment by nylki** [comment]
Additionally to what @wargwargwarg said, adding more danger to plundering may help, eg. self destruct (random self destruct is already implemented afaik) that damages your ship, or danger to your crew for looting instead of only when trying to capture.

**Comment by jmhorjus** [comment]
The exact amount of hull damage on a ship probably shouldn't factor in, since it is very difficult to control this and it could easily go either up or down if the ship is assisted and disabled again.  Other than that, interesting ideas - I encourage you to try implementing some of them and ask questions if you run into trouble.  

The "looting takes time" idea could potentially be made really interesting by just having it take real time in the game.  Currently the entire boarding process is instantaneous from the perspective of the actual space battle happening around you.  What if every time you clicked "Take" on something, the regular space happenings in the system continued for a few seconds (missiles flying, other ships maybe attacking you...or maybe all is quiet if you already defeated all the enemies in the system and no new ones are arriving).  If you get the interface and everything else figured out for that it could be a very engaging and "realistic" risk/reward kind of situation.  
If the time it took was based on outfit weight (square root function?) it might make you think twice about whether you want to try removing a Korath Worldship's reactor while three more of them are bearing down on you. :)

**Comment by Wrzlprnft** [comment]
That idea just came in time. ba-dum-ts. Good one anyway.

**Comment by jmhorjus** [comment]
"Oh, s*** another Wardragon! Cheese it boys!"  
I'm super excited about this now. GUI design will be key.

**Comment by Wrzlprnft** [comment]
It would basically not make a difference in early game, but make it hareder to get really high-value-end-game-stuff.
Though i think that in any form allied or neutral ships should stop firing at the ship you are boarding, because they would otherwise kill your crew.

**Comment by LocalGod79** [comment]
perhaps rename this issue something more concise - perhaps along the lines of "Enhancement: plundering increments time"

as you've noted, there are already half a dozen issues/pulls pertaining to looting - i would hate for this to be given only a cursory glance because of a lack of effort in the title only.

**Comment by db47h** [comment]
Divinity Original Sin has an interesting take on this (pick pocketing is similar to plundering right?): the amount of stuff you can take is limited by weight (determined in turn by your pick-pocketing skill). May be not as exciting as a time limit but could be a decent fallback.

**Comment by endless-sky** [comment]
This seems like a cool idea, and I agree it would increase the excitement of plundering a ship. But, how would capturing work? Already, capturing is much more lucrative than plundering (unless you're plundering a nigh-uncapturable ship like the Quarg). So, ideally each round of ship capture combat would take time, too.

What I'm picturing is maybe that the boarding panel disappears for a second and the game advances, but with your ship unresponsive and some sort of clock timer or ring / bar showing your progress.

**Comment by jmhorjus** [comment]
Yes, the boarding panel disappearing and a your ship unresponsive while a "progress bar" of some sort fills up sounds exactly right. 

If you agree this is a good idea, would you like me to take a crack at implementing it? 

I'm no GUI artist, but maybe I could leverage the existing "ring bar" code that's used to displaying shield/armor status, and you or someone more skilled in that could improve the appearance afterward if neccessary.

**Comment by brick96** [comment]
More crew on both sides should take more/less time I.e. 20 vs 5 will be more time than 40 vs 5 but faster than 40 vs 10

**Comment by Wrzlprnft** [comment]
Couldn't you just move the two parts in a awesome slide animation apart the outfits on the left, the crew on the right, and disable all buttons? On top of the screen a small window with a red 7-segment digits timer comes in, while the "ring bar" on the ship runs down/completes?

**Comment by jafdy** [comment]
It could also add an option to attack aggressively or attack carefully. Aggressive is faster but careful is safer (to your crew).

**Comment by wargwargwarg** [comment]
I'm thinking when boarding is on, ship commands are disabled
Boarding would be:
- turn based, X boarding versus Y defending, or perhaps up to 5 in each team (taking into account that those hallways are cramped), some 3 seconds per turn, subtracting received damage before next round. 
- combat goes on until defenders are all dead or surrender or I push "disengage immediately" . That's for the cases when the ship is being attacked or when combat goes badly.
- random performance in both crews (instead of 1.5 times X fighters, some of them would have cramps or miss or whatever and do only half damage; some are "in-the-zone" and frag twice as many that turn. Random individual performance based on equipment and class (crew is the normal guy with 2 months boot camp, pirates have a lot of combat experience, marines would have the best training)\
- the aggression level is interesting, also it means gaining an advantage (attack) but loosing some defense (hit point)??

Hm.. 100 hit points per marine, each weapon can kill in one shot in optimal conditions, so 100 damage.
Attacking: 5 crew, attack 100, defense 100, 60% chance to hit something in a fight.
Rifles instead of pistols: +50%
Some grade of armor...+35% defense

Defending: 5 pirates, attack 120, defence 120, 75% chance to hit something (tough guys, ok?)
Rifles, nerve gas: +50% +50% attack
Armor: +35% defense
Fighting in their ship: +40% attack and defense

Attack minus Defense = Damage inflicted. 

Those are the general stats, modified further by the ordered aggression level (can be changed in combat)
Individual stats influenced by the random character of combat (attack and defense randomly calculated between 50% and 150% of their general values.

Perhaps with hit points shown for your own team, and being able to call back a badly hurt sailor and replace him with someone unhurt? Husbanding your crew, so to say.

Perhaps experience, with increases to attack, defense and finally hit points? And paycheck, of course.

**Comment by LocalGod79** [comment]
@wargwargwarg i think the developer's aim is to alter a current game mechanic (to balance it somewhat - adding more risk for the potential profit), rather than to add a mini-game to the overall combat mechanic.

there are plenty of turn based strategy games that employ mechanics similar to those you've described - i feel like your last post's suggestions might bog-down/unbalance an otherwise mostly fluid and balanced system.

**Comment by Amazinite** [comment]
This mechanic was never added to the game, but has been milestoned for 1.0.0. I'm reopening it for the sake of discussion; is this a mechanic that we should add to the game? Or would it make plundering (and capturing) too punishing? (Right now it's easy to board a ship in the middle of combat, plunder from or capture it, and leave, whereas this change would make it so that in order to safely plunder/capture, you must get rid of all hostiles, and if at any point an enemy enters the system you'll likely have to abort the plundering/capturing in order to survive.)

**Comment by jmhorjus** [comment]
It doesn't have to take such a long time that any hostile in the system will kill you before it's done. If it takes a very long time then it will also be boring for the player to wait for. Just a short period of vulnerability to discourage looting in the middle of really big fights. Maybe between 1 and 5 seconds depending on the size of the looting/capturing involved. 

p.s. my offer to implement this from 2016 is expired.

**Comment by EligerGaming** [comment]
Here’s my thought on the matter. Time should pass while you are boarding their ship. You can chose to capture, and you would chose to either attack or defend, and then everything waits for like 1-5 seconds, then you can do another round of fighting. This makes it where if you have roughly the same amount of crew on each side, it will take a while, and make big captures, such as using Bactrian to capture a Korath Raide, will take quite a while. If you chose to plunder, then you can select a outfit, then grab it, and the game automatically divides the tonnage of the object by your crew, then that’s how many seconds it takes to grab it. 

I also totally disagree with making certain outfits ‘damaged’ and unplunderable, especially because if your Hull is damaged, your outfits work fine, so it makes no sense to make it damaged when your plundering, and not damaged before you boarded it. 

Crew protecting vital outfits makes a quote a bit of sense. You could have a certain outfits that are deemed important, and a portion of the crew is split between the different vital outfits depending on tonnage or importance. Wargwargwarg also mentioned aggressiveness. While I disagree with aggressiveness effecting how powerful is the crew in terms of fighting, it would be a cool idea to determine what the crew of the ship your boarding are willing to give up. A merchant freighter crew is much less willing to protect their outfits and cargo, because they care more for their lives then protecting their stuff, and they are usually in a friendly, populated place, so anyone that passes by can save them. But an alien faction/species such as the Korath exiles are naturally more aggressive, and each ship needs to pretty much be its own self sustaining colony, so would not only go as far as protecting their vital outfits, but possibly even some of their lesser outfits or even cargo.
